Update pages about links for content and for design (#314)
wpaccessibility/wp-a11y-docs: docs/topics/design/links/index.md

---
title: Styling links
layout: default
parent: Design & user experience
description: When to underline links and when not.
nav_order: 3
contributors:
  - Rian Rietveld
---

# Styling links

> When creating styling for links, the following is best practice for usability and accessibility:

Reviewer: I'm not a fan of paragraphs that exist *only* to tell the user what's going to follow them. The heading should already suffice for that. Maybe a better introductory paragraph would generalize the principles that follow? e.g.

"Best practices for styling links are intended to ensure that links are easily distinguishable from other text on the page. This can be through the design of the link itself, or through the design and structure of the section containing the link, when the links are in a block of navigational content."

> - Underlining links in **paragraphs of text** increases the usability of your website. On the web this is a universal convention that underlined text is a link. 

Reviewer: Tiny hesitation about the word "paragraph", as it could be used to imply that we're only referring to links inside a `p` element. Could this be "blocks of text", instead?

- **Grouped links** in sidebars, footers, or in navigation menus frequently don’t need to be underlined. If it is obvious in context that text is a link, underlining is not required.

> - Use underlining of text only for links. Underlining makes people think something is clickable, if they click it and nothing happens, they'll probably be confused and frustrated. 

Reviewer: ```suggestion
- Use underlining of text **only** for links. Underlining makes people think your text is clickable -- if they click it and nothing happens, they may be confused or frustrated. 
```

Joe Dolson wrote [Why WordPress Themes Require Underlines on Links](https://www.joedolson.com/2025/10/why-wordpress-themes-require-underlines-on-links/) with the reasons why underlining is the best way to indicate links in the content.

For styling hover and focus states of links, read [Styling focus and hover states of interactive HTML elements]({{site.baseurl}}/docs/topics/design/focus-hover/).

> When you custom style the underlining with CSS, make sure the underlining has a color contrast ratio of at least 3:1 against its background color.

Reviewer: ```suggestion
When you create your underlining styles with CSS, make sure the underline has a color contrast ratio of at least 3:1 against its background color.
```

{: .callout .dont }

> **Don't**: remove the underlining from links and use only a different color. Color-blind visitors (about 8%) may miss the link.   

Reviewer: ```suggestion
**Don't**: remove the underlining from links to only use color. Color-blind visitors (about 8% of the population) may miss the link.   
```

I love cats, so I watch <a href="#" style="text-decoration: none; color: darkgreen">cat videos</a> on YouTube.

{: .callout .do }
**Do**: underline links in the content, everyone gets that it's a link.     

> I love cats, so I watch <a href="#" style="color: darkgreen">cat videos</a> on YouTube.

Reviewer: This is something I just thought of while reviewing this....

If we use `<a href="#">` for example links, then if anybody clicks on them, the page position will jump to the top of the page. If we use self-referencing anchors, then the position will still jump, but it'll keep the current focus position in the viewport. E.g. `<a href="#self" id="self">Test</a>`

Should we change those globally?

Author: Yes! Thanks a nice solution.
Let's do this in a separate PR.

## Resources

### Related WCAG Success Criteria for styling links

- [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG22/quickref/#use-of-color) (level A).
- [1.4.11 Non-text Contrast](https://www.w3.org/WAI/WCAG22/quickref/#non-text-contrast) (level AA).

### Other resources

- [Underline Your Links!](https://theadminbar.com/accessibility-weekly/underline-your-links/), by Amber Hinds on the Admin Bar.
- [Why WordPress Themes Require Underlines on Links](https://www.joedolson.com/2025/10/why-wordpress-themes-require-underlines-on-links/), by Joe Dolson.
- [On Link Underlines](https://adrianroselli.com/2016/06/on-link-underlines.html), by Adrian Roselli.
- [Designing Better Links UX](https://smart-interface-design-patterns.com/articles/links-ux/) on Smart Interface Design Patterns.
- [Links and Hypertext - Underlining](https://webaim.org/techniques/hypertext/link_text#underlining), on WebAIM.
- [Guidelines for Visualizing Links](https://www.nngroup.com/articles/guidelines-for-visualizing-links/), by the NNGroup.

> - [Link styling: Underline your links. Test with night mode. Test in greyscale](https://nicsteenhout.substack.com/p/link-styling-underline-your-links), by Nic Steenhout.

Reviewer: ```suggestion
- [Link styling: Underline your links. Test with night mode. Test in greyscale](https://nicolas-steenhout.com/underline-your-links/), by Nic Steenhout.
```

Reviewer: Let's link to Nic's own site for this.
